text/2540-formal-function-parameter-attributes.md

```diff
@@ -0,0 +1,242 @@
+- Feature Name: formal_function_param_attrs
+- Start Date: 2018-10-14
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+[summary]: #summary
+
+This RFC proposes to allow attributes in formal function parameter position.
+
+# Motivation
+[motivation]: #motivation
+
+Having attributes on formal function parameters allows for certain different use cases.
+
+## Example: Handling of unused parameter
+
+In today's Rust it is possible to prefix the name of an identifier to silence the compiler about it being unused.
+With attributes in formal function parameter position we could have an attribute like `#[unused]` that explicitely states this for a given parameter.
+
+```rust
+fn foo(#[unused] bar: u32) -> bool;
```

Comment:
I actually had to use `[[gnu::unused]]`/`[[maybe_unused]]` in C++ and it's *so* much worse than the underscore convention in Rust.

Reply:
Well I think we should leave the underscore convention as it is for now. It was just an idea I had and is not a part of this RFC.

Reply:
I have made it more clear (I hope) that `#[unused]` is just a hypothetical idea around this RFC and not part of the proposal and not a proposal at all.